src/framework/operations.hpp

```diff
@@ -83,6 +86,7 @@ struct Op {
   OpType type;                    // operation type identifier
   std::string name;               // operation name
   reg_t qubits;                   //  qubits operation acts on
+  std::vector<reg_t> qubitss;     //  list of qubits for matrixes
```

Comment:
I would move this down to be above mats and call it:

```
// List of registers
std::vector<reg_t> regs;
```

Since it could be used by other operations (like the multiplexer @jagunnels is working on)